src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorComponentsWeb-CSharp/RazorComponentsWeb-CSharp.App/Startup.cs

```diff
@@ -8,8 +8,7 @@ public class Startup
     {
         public void ConfigureServices(IServiceCollection services)
         {
-            // Since Blazor is running on the server, we can use an application service
-            // to read the forecast data.
+            // Example of a data service
             services.AddSingleton<WeatherForecastService>();
         }
```

Comment:
TBH it's a bit quirky having the data service in the `.App` project.

I think people will ask how to make calls from the `.App` project to `.Server`, but you can't do that directly because the project reference is in the opposite direction. You can do it *indirectly* by declaring `IMyService` in `.App` and configuring an implementation in DI in `.Server`, but we probably don't want to do that in the project because (1) it's more complicated and (2) just as many people would want the implementation in `.App` anyway.

Longer term, this ambiguity-of-goals might push us towards a single-project solution, which would eliminate the question entirely.

Reply:
Not entirely sure if this is the road to take. If we create 1 project we're basically hooking ourselves into the Razor Components flow. It would make sense to have 3 projects
- App :Real App which can be easily swapped to be used as Client Side Blazor instead of being hosted as Razor Components.
- Shared : Models used in the .App and .Server
- Server : The host for a Razor or Blazor app.
--- 
If we later intend to switch from the chattyness of Razor Components to a .wasm, it should be possible with the aforementioned solution structure.